lib/ecto/changeset.ex

```diff
@@ -1629,21 +1629,21 @@ defmodule Ecto.Changeset do
 
   defp wrong_length(_type, value, value, _opts), do: nil
   defp wrong_length(:string, _length, value, opts), do:
-    {message(opts, "should be %{count} character(s)"), count: value, validation: :length, is: value}
+    {message(opts, "should be %{count} character(s)"), count: value, validation: :length, type: :is}
```

Comment:
I think we should only use `:type` for actual data types. I would also think changing this would be backwards incompatible if we except users to use these values.